docs/styleguides/python.md

```diff
@@ -0,0 +1,173 @@
+# DCOS Python Style Guide
+###### Version 0.1
+###### April-6-2017
+
+For Python source code within DCOS, we prefer to keep Google Python Style Guide Revision 2.59 as the base and **extend** it with
+some changes that will make DCOS codebase look modern, consistent and easy to follow.
+
+##### Rationale
+
+With any style guide, the primary requirement is to adopt one and follow it consistently. With our choices below, we can use tools available in the ecosystem to our advantage, enforce consistency in the codebase, creating and publishing automatically generated documentation for our code.
+
+##### Versioned document
+
+Styleguide _conventions_ are not set in stone. We can improve upon this or remove inconvenient styles by creating a PR, discussing the merits or demerits to arrive at a consensus.
+
+#### Base Style Guide
+
+* [Google Python Style Guide Revision 2.59](https://google.github.io/styleguide/pyguide.html)
+
+#### Summary of Style Requirements DCOS which extend Base Style Guide and are specific to DCOS.
+
+* Import Requirements
+    * Adopt the base style guide preferences.
+
+        ```
+        Use import x for importing packages and modules.
+        Use from x import y where x is the package prefix and y is the module name with no prefix.
+        Use from x import y as z if two modules named y are to be imported or if y is an inconveniently long name.
+        ```
+    * Use `__init__.py` only to represent python packages.
+        * Avoid importing of objects in `__init__.py` for readability.
+* Import formatting
+
+    * The types of imports should be separated by single line and order is should be.
+        * standard library imports.
+        * third-party imports.
+        * application specific imports.
+
+    * The `from` style import and `import` statements should be separated by single line too.
```

Comment:
@vespian  - yes, the extreme case you pointed out holds true.
The median case, seems more like

```
# stdlib imports
import stdlibA
import stdlibB
import stdlibC

from stdlibC import fooA

# thirdparty
import thirdpartyA
import thirdpartyB
import thirdpartyC

from thirdpartyD import fooB
from thirdpartyE import fooB as fooBE

# app specific
import appspecificA

from appspecificB import fooC

```

In those cases, it seems to me, that separating types of import gives greater benefits for reading (and collapsing with your editor configuration).

* Readability and collapsibility of similar import lines it the reason for newline between types.
* isort supports this too: It calls this as [lines_between_types](https://github.com/timothycrosley/isort/blob/develop/isort/settings.py#L96)